solutions/swb-app/src/dataSetRoutes.ts

```diff
@@ -86,7 +83,7 @@ export function setUpDSRoutes(router: Router, dataSetService: DataSetPlugin): vo
 
   // share dataset
   router.post(
-    '/datasets/:id/share',
+    '/projects/:projectsId/datasets/:id/share',
     wrapAsync(async (req: Request, res: Response) => {
       if (req.params.id.match(uuidWithLowercasePrefixRegExp(resourceTypeToKey.dataset)) === null) {
```

Comment:
I see two missing validations here:
1. Is the project Id valid?
2. does the dataset requested belong to that project.

If the project Id is in the request body, shouldn't it be taken out and the URL parameter used instead?